src/main/java/org/whaka/mock/EventCollector.java

```diff
@@ -0,0 +1,86 @@
+package org.whaka.mock;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.mockito.Matchers;
+import org.mockito.MockSettings;
+import org.mockito.Mockito;
+import org.mockito.invocation.DescribedInvocation;
+
+/**
+ * Collect arguments from the called method (may be used with listeners).
+ *
+ * @param <Target>  the target class
+ * @param <Event>  the event type
+ */
+public class EventCollector<Target, Event> {
+
+	private final List<Event> events = Collections.synchronizedList(new ArrayList<>());
+	private final Target target;
+
+	@SafeVarargs
+	@SuppressWarnings("unchecked")
+	public EventCollector(Class<Target> targetClass, BiConsumer<Target, Event> targetMethod, Predicate<Event>... filters) {
+		this(targetClass, targetMethod, new HashSet<>(Arrays.asList(filters)));
```

Comment:
`@SuppressWarnings("unchecked")` is unneccessary if `@SafeVarargs` is added. Please remove it.